### PR TITLE
Added Mean and Scale compatibility to OV RT

### PR DESCRIPTION
**Pull request by ujjayant-kadian**
Changed ov_backend.hpp so that now OV RT supports the input of mean and scale as well.

### CODE REVIEW

**Comment by openmodelzooci** [comment]
Can one of the admins verify this patch?

**Comment by TolyaTalamanov** [comment]
> Changed ov_backend.hpp so that now OV RT supports the input of mean and scale as well.

I guess these changes impact `OpenVINO` usage not `ONNXRT`

File: demos/common/cpp_gapi/utils_gapi/include/utils_gapi/ov_backend.hpp [hunk 1]
```diff
@@ -32,10 +32,25 @@ struct BackendApplicator<ExecNetwork,
                 config.deviceName                    // device specifier
             }.cfgNumRequests(config.maxAsyncRequests)
             .cfgPluginConfig(config.getLegacyConfig());
-        // E#96606
-        if (!config.mean_values.empty() || !config.scale_values.empty()) {
-            throw std::runtime_error(backend.name + " backend doesn't support neither `mean_values` nor `scale_values` at the moment");
+            
+        if (!config.mean_values.empty() && !config.scale_values.empty()) {
+            std::vector<float> means;
+            split(config.mean_values, ' ', means);
```
Comment (TolyaTalamanov) [inline demos/common/cpp_gapi/utils_gapi/include/utils_gapi/ov_backend.hpp:38]: I don't think we need to parse `string` into `vector<float>` in every backend we have, it should be generic.
Let `BackendsConfig` holds `vector<float>` instead of `string` and transform `string` to `vector` somewhere on application side (main.cpp).

https://github.com/openvinotoolkit/open_model_zoo/blob/fa567dd88238d975d8b2b39032147922137b71ad/demos/common/cpp_gapi/utils_gapi/include/utils_gapi/backend_description.hpp#L27-L33

File: demos/common/cpp_gapi/utils_gapi/include/utils_gapi/ov_backend.hpp [hunk 2]
```diff
@@ -32,10 +32,25 @@ struct BackendApplicator<ExecNetwork,
...
+            split(config.scale_values, ' ', scales);
+
+            if (means.size() != 3  || scales.size() != 3) {
+                throw std::runtime_error("`mean_values` and `scale_values` must be 3-components vectors "
+                                         "with a space symbol as separator between component values");
+            }
+            net.cfgMean(means).cfgScale(scales);
         }
+        else if (!config.mean_values.empty() && config.scale_values.empty() || config.mean_values.empty() && !config.scale_values.empty() ) {
+            throw std::runtime_error(backends.front().name + " requires both `mean_values` and 'scale_values' "
```
Comment (TolyaTalamanov) [inline demos/common/cpp_gapi/utils_gapi/include/utils_gapi/ov_backend.hpp:50]: `OV` backends doesn't require both to be presented (it has two separate methods)

File: demos/common/cpp_gapi/utils_gapi/include/utils_gapi/ov_backend.hpp [hunk 3]
```diff
@@ -32,10 +32,25 @@ struct BackendApplicator<ExecNetwork,
...
-        // E#96606
-        if (!config.mean_values.empty() || !config.scale_values.empty()) {
-            throw std::runtime_error(backend.name + " backend doesn't support neither `mean_values` nor `scale_values` at the moment");
+            
+        if (!config.mean_values.empty() && !config.scale_values.empty()) {
+            std::vector<float> means;
+            split(config.mean_values, ' ', means);
+
+            std::vector<float> scales;
+            split(config.scale_values, ' ', scales);
```
Comment (TolyaTalamanov) [inline demos/common/cpp_gapi/utils_gapi/include/utils_gapi/ov_backend.hpp:41]: These two lines can be part of `main.cpp` not to repeat them for every backend params

**Review by TolyaTalamanov — CHANGES_REQUESTED** [review]
Great job! 👍 
Some minor changes are required

File: demos/common/cpp_gapi/utils_gapi/include/utils_gapi/ov_backend.hpp (hunk 1 above)
Comment (ujjayant-kadian) [reply demos/common/cpp_gapi/utils_gapi/include/utils_gapi/ov_backend.hpp:38]: So here I should change BackendsConfig, so that they hold vector values instead of strings. And the conversion of string to vector will happen at the application side.

File: demos/common/cpp_gapi/utils_gapi/include/utils_gapi/backend_description.hpp [hunk 4]
```diff
@@ -26,8 +26,8 @@ struct BackendDescription {
 
 struct BackendsConfig: ModelConfig {
     BackendsConfig(const ModelConfig &src,
-                   const std::string &mean_values = "",
-                   const std::string &scale_values = "");
+                   const std::vector<float> &mean_values = "",
```
Comment (TolyaTalamanov) [inline demos/common/cpp_gapi/utils_gapi/include/utils_gapi/backend_description.hpp:29]: Looks `vector<float>` cannot be initialized by `string`

File: demos/common/cpp_gapi/utils_gapi/include/utils_gapi/ov_backend.hpp [hunk 5]
```diff
@@ -32,10 +32,14 @@ struct BackendApplicator<ExecNetwork,
...
-        if (!config.mean_values.empty() || !config.scale_values.empty()) {
-            throw std::runtime_error(backend.name + " backend doesn't support neither `mean_values` nor `scale_values` at the moment");
+
+        if(!config.mean_values.empty()){
+            net.cfgMean(config.mean_values);
+        }
+        if(!config.scale_values.empty()) {
+            net.cfgScale(config.scale_values);
         }
+
```
Comment (TolyaTalamanov) [inline demos/common/cpp_gapi/utils_gapi/include/utils_gapi/ov_backend.hpp:42]: Unnecessary change

File: demos/common/cpp_gapi/utils_gapi/include/utils_gapi/ov_backend.hpp [hunk 6]
```diff
@@ -32,10 +32,14 @@ struct BackendApplicator<ExecNetwork,
                 config.deviceName                    // device specifier
             }.cfgNumRequests(config.maxAsyncRequests)
             .cfgPluginConfig(config.getLegacyConfig());
-        // E#96606
-        if (!config.mean_values.empty() || !config.scale_values.empty()) {
-            throw std::runtime_error(backend.name + " backend doesn't support neither `mean_values` nor `scale_values` at the moment");
+
+        if(!config.mean_values.empty()){
```
Comment (TolyaTalamanov) [inline demos/common/cpp_gapi/utils_gapi/include/utils_gapi/ov_backend.hpp:36]: Looks good, thanks! 👍

File: demos/common/cpp_gapi/utils_gapi/include/utils_gapi/backend_description.hpp (hunk 4 above)
Comment (ujjayant-kadian) [reply demos/common/cpp_gapi/utils_gapi/include/utils_gapi/backend_description.hpp]: Oops, careless error!

**Review by Wovchena — APPROVED** [review]
